cpp/ql/test/query-tests/Security/CWE/CWE-022/semmle/tests/test.c

```diff
@@ -66,22 +66,22 @@ int main(int argc, char** argv) {
 
   {
     void readFile(const char *fileName);
-    readFile(argv[1]); // BAD
+    readFile(argv[1]); // $ Alert=argv
   }
 
   {
     char buffer[1024];
-    read(0, buffer, 1024);
-    read(0, buffer, 1024);
-    fopen(buffer, "wb+"); // BAD [duplicated with both sources]
+    read(0, buffer, 1024); // $ Source=read_output1
+    read(0, buffer, 1024); // $ Source=read_output2
+    fopen(buffer, "wb+"); // $ Alert=read_output1 $ Alert=read_output2
```

Comment:
Could we keep the "duplicated with both sources" comment somehow? Just for complete transparency.

Reply:
Sure; I thought the expectation comments were enough to convey that we expect two sources to flow here, though.

Reply:
If it was not desirable, it should have used `SPURIOUS`. But sure, I'll add the comment back.

Reply:
Yes, I think the alert for `read_output1` should be marked as `SPURIOUS`.  Yet again labelled sources are proving more accurate than the ad-hoc comments we used before.

Reply:
:+1: